contracts/cw4-group/src/state.rs

```diff
@@ -1,10 +1,14 @@
 use cosmwasm_std::{CanonicalAddr, HumanAddr};
-use cw4::{MEMBERS_KEY, TOTAL_KEY};
-use cw_storage_plus::{Item, Map};
+use cw4::TOTAL_KEY;
+use cw_storage_plus::{snapshot_names, Item, SnapshotMap, SnapshotNamespaces, Strategy};
 
 pub const ADMIN: Item<Option<CanonicalAddr>> = Item::new(b"admin");
 pub const TOTAL: Item<u64> = Item::new(TOTAL_KEY);
-pub const MEMBERS: Map<&[u8], u64> = Map::new(MEMBERS_KEY);
+
+// Note: this must be same as cw4::MEMBERS_KEY but macro needs literal, not const
+pub const MEMBERS: SnapshotMap<&[u8], u64> =
+    SnapshotMap::new(snapshot_names!("members"), Strategy::EveryBlock);
```

Comment:
helper `fn` instead of macro? (I hate macros) :-)

Reply:
Problem is I want `&'static [u8]`
No way I see to do with functions. But if you can write one, I am happy to accept that pr

Reply:
I have also avoided macros strongly. This is the first time I wrote one

Reply:
I see. Not possible without eliminating the references inside `SnapshotNamespaces`.

Reply:
I could take it as 3 different arguments instead of the struct, but they end up in Map and have to be `static so we can keep it as a global const

Reply:
Yes... I was playing with this and it's not so much that it can't be done in a function (it can, using `Vec` instead of slices) but that `Map::new()` is marked `const` and so ir requires `const` parameters.